metadata/metadata-modeshape/src/main/java/com/thinkbiganalytics/metadata/modeshape/BaseJcrProvider.java

```diff
@@ -355,12 +408,79 @@ public void deleteById(PK id) {
         };
     }
 
+    protected StringBuilder startBaseQuery(String... props) {
+        String alias = getEntityAlias();
+        StringBuilder bldr = new StringBuilder("SELECT ");
+        if (props.length == 0) {
+            bldr.append(alias).append(".* ");
+        } else {
+            boolean start = true;
+            for (String prop : props) {
+                if (! start) {
+                    bldr.append(", ");
+                    start = false;
+                }
+                bldr.append(alias).append(".").append(prop);
+            }
+        }
+        return bldr.append(" FROM [").append(getNodeType(getJcrEntityClass())).append("] AS ").append(getEntityAlias()).append(" ");
+    }
+    
+    protected String getEntityAlias() {
+        return "e";
+    }
+
     protected String sanitizeSystemName(String systemName) {
         return INVALID_SYSTEM_NAME_PATTERN.matcher(systemName).replaceAll("_");
     }
 
     protected String sanitizeTitle(String title) {
         return StringEscapeUtils.escapeJava(title);
     }
+    
+    protected void appendJoins(StringBuilder bldr, String filter) {
+        // No joins by default.  Subclasses should override to add any joins needed for sorting and/or filtering.
+    }
+    
+    protected void appendJoins(StringBuilder bldr, Pageable pageable, String filter) {
+        // By default, only adds the joins for the filter.
+        // Subclasses should override to add any joins needed for sorting and/or filtering.
+        appendJoins(bldr, filter);
+    }
+
+    protected void appendFilter(StringBuilder bldr, String filter) {
+        // No filtering by default.  Subclasses should override to filter on any appropriate entity fields.
+    }
+
+    protected void appendOffset(StringBuilder bldr, Pageable pageable) {
+        bldr.append(" LIMIT ").append(pageable.getPageSize()).append(" OFFSET ").append(pageable.getOffset());
+    }
+
+    protected void appendSort(StringBuilder bldr, Pageable pageable) {
+        boolean first = true;
+        
+        if (pageable.getSort() != null && pageable.getSort().iterator().hasNext()) {
+            bldr.append("ORDER BY ");
+            for (Sort.Order order : pageable.getSort()) {
+                if (! first) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Change this condition so that it does not always evaluate to "false" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2583)